libcxx/test/std/contracts/enforce.pass.cpp

```diff
@@ -0,0 +1,27 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20, c++23
+
+// <contracts>
+
+// TODO
+
+// ADDITIONAL_COMPILE_FLAGS: -D_LIBCPP_CONTRACTS_EVALUATION_SEMANTIC_ENFORCE
```

Comment:
Not really needed for this iteration, but I think we need to consider how this works in the modular world. Especially when users want to mix the contract enforcement behaviour on different translation units.

We should not forget to add this new header to modules.

Reply:
Fully agreed. In reality this would be passed to the compiler as e.g. `-fcontracts=<semantic>` so there would be no macro involved. Similarly the `__pre__` and `__post__` macros wouldn't need to exist, I was just roughly emulating what the compiler might do.